src/components/categories/weather/Weather.vue

```diff
@@ -0,0 +1,153 @@
+                thxButton: false,
+                weatherData: '',
+            }
+        },
+        mounted() {
+            this.$root.$on('cityDropdownOnClick', (text) => {
+                this.cityDropdownOnClick(text);
+            });
+            this.$root.$on('showWeather', (data) => {
+                this.time = data[1];
+                this.showWeather(data[0]);
+            });
+            this.$root.$on('event', () => {
+                this.endWeatherTalk();
+            });
+        },
+        methods: {
+            cityDropdownOnClick(value) {
+                this.addUserMessage(`${this.$t('weather.user.chooseCity')} ${value}`);
+                this.city = value;
+                document.getElementById("dropdown-button").remove();
+                this.addBotMessage(this.$t('weather.bot.choiceTime'));
+                this.addDropdownTime();
+            },
+            showWeather(value) {
+                this.addUserMessage(`${this.$t('weather.user.myChoice')} ${value}`);
+                this.addBotMessage(`${this.$t('weather.bot.myPredictions')} ${this.city} ${this.$t('weather.bot.in')} ${value}....`);
+                this.addWeatherMessage();
+                this.thxButton = true;
+
+            },
+            addBotMessage(text) {
+                let ComponentClass = Vue.extend(BotMessage);
+                let instance = new ComponentClass();
+                instance.text = text;
+                instance.$mount();
+                document.getElementById("weather-component").appendChild(instance.$el);
+
+                this.addBotImage()
+
+            },
+            addUserMessage(text) {
+                const ComponentClass = Vue.extend(UserMessage);
+                const instance = new ComponentClass();
+                instance.text = text;
+                instance.$mount();
+                document.getElementById("weather-component").appendChild(instance.$el);
+            },
+            addDropdownTime() {
+                this.timeDropdown = true;
+            },
+            addWeatherMessage() {
+                weatherService.getWeatherData(this.city, this.time).then((weatherData) => {
+                    weatherData.city = this.city;
+                    weatherData.time = this.time;
+                    this.weatherData = weatherData;
+                    let ComponentClass = Vue.extend(WeatherMessage);
+                    let instance = new ComponentClass();
+                    instance.data = weatherData;
+                    instance.$mount();
+                    document.getElementById("weather-component").appendChild(instance.$el);
+                    this.addBotImage();
+                });
+            },
+            endWeatherTalk() {
+
+                const div = document.getElementById('moreDetailsButton');
+                if(div!== null){
+                    div.remove();
+                }
+
+                document.getElementById('endWeatherTalkButton').remove();
```

Comment:
I ponownie, conditional rendering. Wydaje się, że można zastosować we wszystkich przypadkach